crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -1117,6 +1118,491 @@ TEST_P(HMACServiceIndicatorTest, HMACTest) {
+    0x28, 0xb0, 0x80, 0xf7, 0xc0, 0x68
+};
+
+// RFC Test Case 5 repeats the inputs from RFC 5869 Test Case 2; RFC Test Case 6
+// repeats the inputs from RFC 5869 Test Case 3.
+
+const uint8_t kHKDF_ikm_tc7[] = {   // RFC 5869 Test Case 7
+    0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c,
+    0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c, 0x0c
+};
+// Salt for Test Case 7 is not specified (NULL). HKDF will use HashLen 0x00
+// bytes instead.
+const uint8_t kHKDF_info_tc7[] = {0};    // No info
+const uint8_t kHKDF_okm_tc7_sha1[] = {
+    0x2c, 0x91, 0x11, 0x72, 0x04, 0xd7, 0x45, 0xf3, 0x50, 0x0d, 0x63, 0x6a,
+    0x62, 0xf6, 0x4f, 0x0a, 0xb3, 0xba, 0xe5, 0x48, 0xaa, 0x53, 0xd4, 0x23,
+    0xb0, 0xd1, 0xf2, 0x7e, 0xbb, 0xa6, 0xf5, 0xe5, 0x67, 0x3a, 0x08, 0x1d,
+    0x70, 0xcc, 0xe7, 0xac, 0xfc, 0x48
+};
+const uint8_t kHKDF_okm_tc7_sha224[] = {
+    0xca, 0x84, 0x01, 0xe6, 0x45, 0xb3, 0xa5, 0x8e, 0x00, 0x99, 0x28, 0x57,
+    0xfe, 0x00, 0x38, 0xcb, 0x1b, 0xf8, 0xdc, 0x51, 0xed, 0xf0, 0x52, 0x33,
+    0x6c, 0x08, 0xf3, 0xbe, 0xd6, 0x82, 0xc8, 0x3e, 0x77, 0x80, 0x3c, 0xdd,
+    0x16, 0xd1, 0x56, 0xbb, 0x8a, 0x30
+};
+const uint8_t kHKDF_okm_tc7_sha256[] = {
+    0x59, 0x68, 0x99, 0x17, 0x9a, 0xb1, 0xbc, 0x00, 0xa7, 0xc0, 0x37, 0x86,
+    0xff, 0x43, 0xee, 0x53, 0x50, 0x04, 0xbe, 0x2b, 0xb9, 0xbe, 0x68, 0xbc,
+    0x14, 0x06, 0x63, 0x6f, 0x54, 0xbd, 0x33, 0x8a, 0x66, 0xa2, 0x37, 0xba,
+    0x2a, 0xcb, 0xce, 0xe3, 0xc9, 0xa7
+};
+const uint8_t kHKDF_okm_tc7_sha384[] = {
+    0x6a, 0xd7, 0xc7, 0x26, 0xc8, 0x40, 0x09, 0x54, 0x6a, 0x76, 0xe0, 0x54,
+    0x5d, 0xf2, 0x66, 0x78, 0x7e, 0x2b, 0x2c, 0xd6, 0xca, 0x43, 0x73, 0xa1,
+    0xf3, 0x14, 0x50, 0xa7, 0xbd, 0xf9, 0x48, 0x2b, 0xfa, 0xb8, 0x11, 0xf5,
+    0x54, 0x20, 0x0e, 0xad, 0x8f, 0x53
+};
+const uint8_t kHKDF_okm_tc7_sha512[] = {
+    0x14, 0x07, 0xd4, 0x60, 0x13, 0xd9, 0x8b, 0xc6, 0xde, 0xce, 0xfc, 0xfe,
+    0xe5, 0x5f, 0x0f, 0x90, 0xb0, 0xc7, 0xf6, 0x3d, 0x68, 0xeb, 0x1a, 0x80,
+    0xea, 0xf0, 0x7e, 0x95, 0x3c, 0xfc, 0x0a, 0x3a, 0x52, 0x40, 0xa1, 0x55,
+    0xd6, 0xe4, 0xda, 0xa9, 0x65, 0xbb
+};
+
+static const struct HKDFTestVector {
+    // func is the hash function for HMAC to test.
+    const EVP_MD *(*func)(void);
+    const uint8_t *ikm; // Initial Keying Material
+    const size_t ikm_size;
+    const uint8_t *salt;    // Salt
+    const size_t salt_size;
+    const uint8_t *info;    // "Other Info", the sequel to Salt.
+    const size_t info_size;
+    const uint8_t *expected_output;    // Expected Output Keying Material
+    const uint8_t output_len;
+    const FIPSStatus expect_approved;
+} kHKDFTestVectors[] = {
+    // RFC 5869 Test Case 1
+    {
+        EVP_sha1,
+        kHKDF_ikm_tc1, sizeof(kHKDF_ikm_tc1),
+        kHKDF_salt_tc1, sizeof(kHKDF_salt_tc1),
+        kHKDF_info_tc1, sizeof(kHKDF_info_tc1),
+        kHKDF_okm_tc1_sha1, sizeof(kHKDF_okm_tc1_sha1),
+        AWSLC_APPROVED,
```

Comment:
Maybe we should add another test that checks for `AWSLC_NOT_APPROVED` if the `HKDF` function fails

Reply:
Good idea, I'll throw in one using MD5.